src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking.cs

```diff
@@ -14640,6 +14652,228 @@ public S(A a, B b)
                 Diagnostic(ErrorCode.WRN_NullReferenceArgument, "null").WithArguments("a", "S.S(A a, B b)").WithLocation(7, 26));
         }
 
+        [Fact]
+        public void WarningOnConversion_Assignment()
+        {
+            var source =
+@"class Person
+{
+    internal string FirstName { get; set; }
+    internal string LastName { get; set; }
+    internal string? MiddleName { get; set; }
+}
+class Program
+{
+    static void F(Person p)
+    {
+        p.LastName = null; // no warning
+        p.LastName = (string?)null; // warning
```

Comment:
I've removed the comments but the tests should represent the expected behavior now.